azure/services/resourceskus/sku.go

```diff
@@ -128,3 +130,23 @@ func (s SKU) GetCapability(name string) (string, bool) {
 	}
 	return "", false
 }
+
+// HasLocationCapability returns true if the provided resource supports the location capability.
+func (s SKU) HasLocationCapability(name, size, location string) bool {
+	if s.LocationInfo != nil && *s.Name == size {
+		for _, info := range *s.LocationInfo {
+			if info.ZoneDetails != nil && *info.Location == location {
+				for _, zone := range *info.ZoneDetails {
+					if zone.Capabilities != nil {
+						for _, capability := range *zone.Capabilities {
+							if *capability.Name == name {
+								return true
+							}
+						}
+					}
+				}
+			}
+		}
+	}
+	return false
+}
```

Comment:
```suggestion
func (s SKU) HasLocationCapability(name, size, location, zone string) bool {
	if s.LocationInfo != nil && *s.Name == size {
		for _, info := range *s.LocationInfo {
			if info.ZoneDetails != nil && *info.Location == location {
				for _, zone := range *info.ZoneDetails {
					if zone.Capabilities != nil {
						for _, capability := range *zone.Capabilities {
							if *capability.Name == name {
							        if zone.Names != nil {
							                for _, name := zone.Names {
							                        if name == zone {
							                                return true
							                        }
							                }
							        }
								return false
							}
						}
					}
				}
			}
		}
	}
	return false
}
```

Reply:
the diff is the func signature (added zone) and the inner loop on zone.Names, in case that code suggestion is hard to read

Reply:
modified a bit to reduce nesting, but same logic

Reply:
how can we accommodate zone in case of scaleset? i think there we have zone list, so for each zone in the list we have to check the capability?

Reply:
ah yeah, this is a bit tricky. what you said is the safest for sure, but does restrict some cases where the user would technically be able to deploy the VM. however, if they asked for ultrassd and didn't get it e.g. defaulted on, I think it's reasonable to fail the operation, since we can't fulfill their expectation that they will actually be able to get ultrassd.

@CecileRobertMichon or @jackfrancis wdyt?

Reply:
Makes sense to me